### ISSUE TITLE
"meteor run ios" shows a blank page

### ISSUE THREAD

**Issue by sunshineo** [Urigo/angular2-meteor#284]
I'm following the tutorial.
My socially app runs properly in browser.
However when run it in ios simulator it just shows a blank page.
I inspected the cordova build and it shows no js file I expected to build from the ts files.
I downloaded https://github.com/Urigo/meteor-angular2.0-socially and it is the same thing.
If this is still work in progress, I think we need to indicate this in the tutorial.

![nojsfiles](https://cloud.githubusercontent.com/assets/1072766/15625920/8e5d7526-246c-11e6-8572-3b419850be65.jpg)

**Comment by metzc** [comment]
i can confirm this bug. i'm trying to figure this out for a few days now (iOS "white screen of death"). In order to isolate the problem i started a new meteor project which worked on iOS just fine. As soon as i add "angular2-compilers" and ("barbatus:angular2-runtime") and a very basic "app.ts" (exactly like in the README / tutorial) the app fails to load on iOS (emulate or via xcode). no error message is shown on the terminal.

When i debug the emulator app with safari and hit reload from within the Web Inspector i get:
`EXCEPTION: RangeError: Maximum call stack size exceeded.`
along with a long list of this statements:
`get@http://localhost:12840/packages/modules.js?hash=1b42df9a0cbd6ab5e8abc7c7e9cf14b2504c4ca7:45974:57`

Anyway it should be very easy to reproduce.

**Comment by sunshineo** [comment]
Thank you @metzc for confirming this. I think the project is simply not there yet. I hope we can add a note on the tutorial. I may even write a pull request for that.

**Comment by metzc** [comment]
@sunshineo i think the js files are just fine. there is one unified app.js on the app-root, you will find all the code in there. the subfolders just contain the views.

anyway, i found out that it works as soon as the code is hot-reloaded. After one hot-reload you can close and re-open the app and it still works. This is true for both "run ios" and "run ios-device". When building the app ("meteor build ...") it works right away with out an initial hot-reload.

So i can work with this for now. However a fix would be nice anyway of course ;)

**Comment by sunshineo** [comment]
@metzc What exactly do you mean by "hot-reloaded" I tried to modify the code while the ios simulator is running but then the console shows "=> Client modified -- refreshing" and never change.
I tried to kill the app and then start it but no use.

**Comment by metzc** [comment]
yes, thats hot-reload. if it still not works for you, your code has issues. i found that when using cordova with meteor you are much more likely to run into problems (which often cause the app to be stuck in white screen) than on (desktop-)browser. this can have multiple reasons. one of them may be related to angular2-meteor and the way it puts the code together (i suspect that the problem is the order of execution @Urigo). Anyway it should work if your code is structured in a solid way. For me it was removing some left over code files in a folder which were not included anymore in my project. On desktop they didn't cause any trouble, on mobile they were included on the very top of "app.js".

**Comment by sunshineo** [comment]
I hope this get fixed soon

**Comment by barbatus** [comment]
@metzc I just tried TODO demo in ios emulator using Angular2-Meteor master version, and had no any problem. Can you check that master version version works fine for you as well?

**Comment by sunshineo** [comment]
Hello @barbatus , I was following http://www.angular-meteor.com/tutorials/socially/angular2/bootstrapping . I have 2 questions for you. First, can you give us a link to the TODO demo? Second, what do you mean by "using Angular2-Meteor master version"? How do I use it? Do I need to run something different than "meteor npm install --save angular2-meteor @angular/platform-browser-dynamic" ? Do I need to specify "master" somewhere in the command? Thank you very much!

**Comment by barbatus** [comment]
@sunshineo Master means installing this NPM from the github repo, but never mind. This issue is likely related to #273 which means you'll need to fix the way you use zone.js.  BTW, there shouldn't be any js-file in the folder you pointed out - all app JS-files are bundled in `app.js`.

**Comment by barbatus** [comment]
Ok, not sure where is the issue if any since I was not able to reproduce it.
Anyways, if you are using `barbatus:angular2-runtime`, update it to `0.5.0` version which relies on Meteor 1.3's modules to import deps. If you are still facing the issue I'll need a repo to reproduce.

<img width="532" alt="screen shot 2016-06-07 at 1 43 09 pm" src="https://cloud.githubusercontent.com/assets/2824096/15856690/58176230-2cb7-11e6-8d2c-61ca692471aa.png">

**Comment by metzc** [comment]
I tried your meteor 1.3 TODO example which worked.
Updating barbatus:angular2-runtime to 0.5.0 helped in a way that i got an useful error message on mobile now. I tried to access a Meteor.settings value (from settings.json) which worked on browser but failed on ios (another bug?). Anyway, i removed that for now, as everything else is working now.

TLDR: thanks.

**Comment by sunshineo** [comment]
@barbatus 
I still have the problem. Here is what I did:
// update my meteor to lates
curl https://install.meteor.com/ | sh
// clone this project
git clone https://github.com/Urigo/meteor-angular2.0-socially
cd meteor-angular2.0-socially
// do a meteor update
meteor update
// install npm packages
meteor npm install 
// start it
meteor add-platform ios
meteor run ios

I got a blank page.

Where is your TODO app?

**Comment by sunshineo** [comment]
Hi @barbatus , when I run with the ng2-todo-ssr app, I do see a non-white screen. So I guess it is a problem with the meteor-angular2.0-socially app 

I got this error stack with ng2-todo-ssr

it-sea15228-mbp:ng2-todo-ssr ssun$ meteor run ios
[[[[[ ~/Github/ng2-todo-ssr ]]]]]

=> Started proxy.
=> Started MongoDB.
client/bootstrap.ts (1, 25): Cannot find module 'meteor/barbatus:ng2-uni'.
route.ts (1, 26): Cannot find module 'meteor/barbatus:ng2-uni'.
route.ts (4, 1): Cannot find name 'FlowRouter'.
Compiling HTML template: imports/components/task-list.html
Compiling HTML template: imports/components/task.html
Compiling HTML template: imports/app.html
client/bootstrap.ts (1, 25): Cannot find module 'meteor/barbatus:ng2-uni'.
route.ts (1, 26): Cannot find module 'meteor/barbatus:ng2-uni'.
route.ts (4, 1): Cannot find name 'FlowRouter'.
route.ts (1, 26): Cannot find module 'meteor/barbatus:ng2-uni'.
route.ts (4, 1): Cannot find name 'FlowRouter'.
W20160629-18:47:07.603(-7)? (STDERR)
W20160629-18:47:07.732(-7)? (STDERR) /Users/ssun/.meteor/packages/meteor-tool/.1.3.4_1.1ff5caj++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:280
W20160629-18:47:07.733(-7)? (STDERR)            				throw(ex);
W20160629-18:47:07.733(-7)? (STDERR)            				      ^
W20160629-18:47:07.733(-7)? (STDERR) TypeError: undefined is not a function
W20160629-18:47:07.734(-7)? (STDERR)     at Object.<anonymous> (/Users/ssun/Github/ng2-todo-ssr/node_modules/angular2-universal/dist/node/helper.js:3:25)
W20160629-18:47:07.734(-7)? (STDERR)     at Module._compile (module.js:456:26)
W20160629-18:47:07.735(-7)? (STDERR)     at Object.Module._extensions..js (module.js:474:10)
W20160629-18:47:07.735(-7)? (STDERR)     at Module.load (module.js:356:32)
W20160629-18:47:07.735(-7)? (STDERR)     at Module.Mp.load (/Users/ssun/.meteor/packages/babel-compiler/.6.8.3.npg8y8++os+web.browser+web.cordova/npm/node_modules/reify/node/runtime.js:16:23)
W20160629-18:47:07.735(-7)? (STDERR)     at Function.Module._load (module.js:312:12)
W20160629-18:47:07.736(-7)? (STDERR)     at Module.require (module.js:364:17)
W20160629-18:47:07.736(-7)? (STDERR)     at require (module.js:380:17)
W20160629-18:47:07.736(-7)? (STDERR)     at Object.<anonymous> (/Users/ssun/Github/ng2-todo-ssr/node_modules/angular2-universal/dist/node/platform/dom/node_dom_renderer.js:29:16)
W20160629-18:47:07.736(-7)? (STDERR)     at Module._compile (module.js:456:26)
